dev-client/src/components/buttons/TextButton.tsx

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright © 2024 Technology Matters
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see https://www.gnu.org/licenses/.
+ */
+
+import {useState} from 'react';
+import {AccessibilityProps} from 'react-native';
+import {PressableProps} from 'react-native-paper/lib/typescript/components/TouchableRipple/Pressable';
+
+import {Button} from 'native-base';
+
+import {Icon, IconName} from 'terraso-mobile-client/components/icons/Icon';
+
+export type TextButtonType = 'default' | 'error';
+
+export type TextButtonProps = {
+  label: string;
+  type?: TextButtonType;
+  role: AccessibilityProps['role'];
+  leftIcon?: IconName;
+  rightIcon?: IconName;
+  disabled?: boolean;
+  onPress?: PressableProps['onPress'];
+};
+
+export const TextButton = ({
+  label,
+  type = 'default',
+  role,
+  leftIcon,
+  rightIcon,
+  disabled,
+  onPress,
+}: TextButtonProps) => {
+  const [pressed, setPressed] = useState(false);
+  const colors = disabled ? COLORS.disabled : COLORS[type];
+
+  return (
+    <Button
```

Comment:
The styling here is very cumbersome -- am open to an alternative implementation, but we will also be revisiting all buttons code in the future so we don't need to make a definitive call right now.